src/fsck/fsck.c

```diff
@@ -414,7 +414,7 @@ static int run(int argc, char *argv[]) {
         exit_status = wait_for_terminate_and_check("fsck", pid, WAIT_LOG_ABNORMAL);
         if (exit_status < 0)
                 return exit_status;
-        if (exit_status & ~1) {
+        if ((exit_status & ~FSCK_ERROR_CORRECTED) != 0) {
```

Comment:
Do we really need the `!= 0` part? I think it just distracts.

Reply:
i was tempted to, but our coding style we should drop it only for boolean-like stuff, but this is kinda not so much anymore, after all we mask out the "corrected" flag and leave all the other bits on, which are not booleans as a whole...

but actually, looking at this, I think this should be rewritten as:

```c
if ((exit_status & ~FSCK_ERROR_CORRECTED) == FSCK_SUCCESS)) …
```